v3/mathjax3-ts/core/MathItem.ts

```diff
@@ -0,0 +1,331 @@
+/*************************************************************
+ *
+ *  Copyright (c) 2017 The MathJax Consortium
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/**
+ * @fileoverview  Implements the interface and abstract class for MathItem objects
+ *
+ * @author dpvc@mathjax.org (Davide Cervone)
+ */
+
+import {MathDocument} from './MathDocument.js';
+import {InputJax} from './InputJax.js';
+import {OptionList} from '../util/Options.js';
+import {MmlNode} from './MmlTree/MmlNode.js';
+
+/*****************************************************************/
+/*
+ *  The Location gives a location of a position in a document
+ *  (either a node and character position within it, or
+ *  an index into a string array, the character position within
+ *  the string, and the delimiter at that location).
+ */
+
+export interface Location {
+    i?: number;
+    n?: number;
+    delim?: string;
+    node?: Text | Element;
+}
+
+/*****************************************************************/
+/*
+ *  The Metrics object includes tht data needed to typeset
+ *  a Mathitem.
+ */
+
+export interface Metrics {
+    em: number;
+    ex: number;
+    containerWidth: number;
+    lineWidth: number;
+    scale: number;
+}
+
+/*****************************************************************/
+/*
+ *  The BBox object contains the data about the bounding box
+ *  for the typeset element.
+ */
+
+export interface BBox {
+    // will be defined later
+}
+
+/*****************************************************************/
+/*
+ *  The MathItem interface
+ *
+ *  The MathItem is the object that holds the information about a
+ *  particular expression on the page, including pointers to
+ *  where it is in the document, its compiled version (in the
+ *  internal format), its typeset version, its bounding box,
+ *  and so on.
+ */
+
+export interface MathItem {
```

Comment:
No, `ProtoItem` is a separate type whose data is used to create the actual `MathItem` alter.